### ISSUE TITLE
Web Workers support

### ISSUE THREAD

**Issue by mcharytoniuk** [mozilla/nunjucks#455]
When I try to import nunjucks under Web Worker I receive the message (I can't provide exact line number because nunjucks is a part of a bundled script in my case): 

Firefox:
```JavaScript
TypeError: loaders is undefined
```
```JavaScript
var lib = modules["lib"];
var env = modules["environment"];
var compiler = modules["compiler"];
var parser = modules["parser"];
var lexer = modules["lexer"];
var runtime = modules["runtime"];
var Loader = modules["loader"];
var loaders = modules["loaders"];
var precompile = modules["precompile"];

nunjucks = {};
nunjucks.Environment = env.Environment;
nunjucks.Template = env.Template;

nunjucks.Loader = Loader;
HERE ---> nunjucks.FileSystemLoader = loaders.FileSystemLoader;
nunjucks.WebLoader = loaders.WebLoader;

nunjucks.compiler = compiler;
nunjucks.parser = parser;
nunjucks.lexer = lexer;
nunjucks.runtime = runtime;
```

Chrome:
```JavaScript
Uncaught Error: Uncaught TypeError: Cannot read property 'FileSystemLoader' of undefined
```

Actually nunjucks picks `node-workers` inside WebWorker.

It would be nice to support web workers so it would be possible to render templates in the separate non-blocking thread and just return the result.

**Comment by mcharytoniuk** [comment]
I did overcome this by replacing such strings in the nunjucks source code during my gulp build but it is obviously a hack:

```JavaScript
.pipe(replace("typeof window === 'undefined' || window !== this", 'false'))
.pipe(replace("window.", "self."))
```

I think web workers should work out of the box.

**Comment by jlongster** [comment]
Can you try the files from master? That part of the code has been refactored and it's a lot more robust now. That check is gone. It should work now, although precompiled templates won't work...

**Comment by retorquere** [comment]
I'm hitting the same error. What file should I use instead of the one I download from http://mozilla.github.io/nunjucks/files/nunjucks.js ?

**Comment by jlongster** [comment]
@retorquere try using that file now. I was hoping you could try from master, but master just got released at 2.0, so the file there is the new version. I think it should work now.

**Comment by retorquere** [comment]
@jlongster I no longer use nunjucks, a.o. because it prevents my extension from qualifying for auto
signing at AMO. Thanks for the effort though.
